x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/Mapper.java

```diff
@@ -61,27 +63,29 @@ private static class SimpleExecMapper extends MapExecRule<LogicalPlan> {
 
         @Override
         protected PhysicalPlan map(LogicalPlan p) {
-
-            if (p instanceof Sequence s) {
-                List<List<Attribute>> keys = new ArrayList<>(s.children().size());
+            if (p instanceof AbstractJoin join) {
+                List<List<Attribute>> keys = new ArrayList<>(join.children().size());
                 List<PhysicalPlan> matches = new ArrayList<>(keys.size());
 
-                for (KeyedFilter keyed : s.queries()) {
+                for (KeyedFilter keyed : join.queries()) {
                     keys.add(Expressions.asAttributes(keyed.keys()));
                     matches.add(map(keyed.child()));
                 }
 
-                return new SequenceExec(
-                    p.source(),
-                    keys,
-                    matches,
-                    Expressions.asAttributes(s.until().keys()),
-                    map(s.until().child()),
-                    s.timestamp(),
-                    s.tiebreaker(),
-                    s.direction(),
-                    s.maxSpan()
-                );
+                if (p instanceof Sequence sequence) {
```

Comment:
Nit, switch the check: check on Sample and return it otherwise fallback to the existing code - less code modified.